src/test/java/fakes/UserControllerTest.java

```diff
@@ -65,15 +101,28 @@ void withValidInexistentUsername_shouldAddUserToDB__MOCKITO() {
 
       // Tipp: Du kannst prüfen, ob der User hinzugefügt wurde,
       //  indem du prüfst wie of die Methode Database.addUser aufgerufen wurde.
-      // TODO implement test
+      // -> implement test
+      /* ARRANGE */
+      UserValidator uv = mock(UserValidator.class);
+      FileDatabase db = mock(FileDatabase.class);
+      UserController uc = new UserController(uv, db);
+      User user = new User("kalua");
+
+
+      /* ACT */
+      Message result = uc.create(user);
+      verify(db, times(1)).addUser(any());
```

Comment:
Ist dieser Test grün? Denn eigentlich sollten doch die beiden UserValidator Methoden false retournieren (default Rückgabewert für boolean Methoden, falls nicht anders spezifiziert via doReturn).

Reply:
Ja, bei mir ist der grün ;-)
Ich glaube ich weiss auf was du raus willst. die Methoden isValidUsername und doesUsernameExist müssten mit doReturn(true)... resp. doReturn(false) angewiesen werden, die entsprechende Rückgabe zu machen, damit im Controller der Check entsprechend funktioniert.
Für isValidUsername funktioniert das. Nicht jedoch für doesusernameExist. Da gibts ein Error, der mich leicht überfordert: "Misplaced oder misused argument matcher detected..."

Reply:
Hilft dir der erste Google Treffer weiter https://stackoverflow.com/questions/23273230/misplaced-argument-matcher-detected-here-you-cannot-use-argument-matchers-outsi ? Kann es sein, dass du z.B. anyString() ausserhalb des Mockito Kontext genutzt hast? Also nicht für ein doReturn / verify, sondern für einen echten Aufruf deines SUT?